src/GraphQl/Type/TypeBuilder.php

```diff
@@ -104,27 +107,25 @@ public function getResourceObjectType(?string $resourceClass, ResourceMetadataCo
             'name' => $shortName,
             'description' => $operation->getDescription(),
             'resolveField' => $this->defaultFieldResolver,
-            'fields' => function () use ($resourceClass, $operation, $operationName, $resourceMetadataCollection, $input, $wrapData, $depth, $ioMetadata) {
+            'fields' => function () use ($resourceClass, $operation, $resourceMetadataCollection, $input, $wrapData, $depth, $ioMetadata) {
                 if ($wrapData) {
                     $queryNormalizationContext = $this->getQueryOperation($resourceMetadataCollection)?->getNormalizationContext() ?? [];
 
                     try {
-                        $mutationNormalizationContext = $operation instanceof Mutation || $operation instanceof Subscription ? ($resourceMetadataCollection->getOperation($operationName)->getNormalizationContext() ?? []) : [];
+                        $mutationNormalizationContext = $operation instanceof Mutation || $operation instanceof Subscription ? ($operation->getNormalizationContext() ?? []) : [];
                     } catch (OperationNotFoundException) {
                         $mutationNormalizationContext = [];
                     }
                     // Use a new type for the wrapped object only if there is a specific normalization context for the mutation or the subscription.
                     // If not, use the query type in order to ensure the client cache could be used.
                     $useWrappedType = $queryNormalizationContext !== $mutationNormalizationContext;
 
-                    $wrappedOperationName = $operationName;
+                    $wrappedOperation = $operation;
 
                     if (!$useWrappedType) {
-                        $wrappedOperationName = $operation instanceof Query ? $operationName : 'item_query';
+                        $wrappedOperation = $resourceMetadataCollection->getGraphQlOperation(null, forceCollection: $operation instanceof Query);
```

Comment:
Why using a named argument here and not elsewhere?

Reply:
I want to start using them I can revert for the codestyle

Reply:
Personally I think it's better like this indeed.